controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -717,7 +715,11 @@ func (r *BareMetalHostReconciler) getPreprovImage(info *reconcileInfo, formats [
 			},
 			Spec: expectedSpec,
 		}
-		controllerutil.SetControllerReference(info.host, &preprovImage, r.Scheme())
+		err = controllerutil.SetControllerReference(info.host, &preprovImage, r.Scheme())
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to set controller reference for PreprovisioningImage")
```

Comment:
nit: no deprecated errors.Wrap please

Reply:
ow there are plenty of it in the current code, this was a copy paste effect .

Reply:
I will try to address this as a whole for the whole repo in a followup if thats ok.

Reply:
I have changed it anyhow here, no point adding new entries of deprecated errors.Wrap as you said.